lib/Model/Record/ClassRecord.php

```diff
@@ -33,38 +17,24 @@ class ClassRecord
      */
     private $implemented = [];
 
-    /**
-     * @var string
-     */
-    private $filePath;
-
-    /**
-     * @var ByteOffset
-     */
-    private $start;
+    public static function fromName(string $name): self
+    {
+        return new self(FullyQualifiedName::fromString($name));
```

Comment:
Careful with `self`, I would suggest declaring the class final or to use `static` if it make sens to extend it.

Reply:
yeah wanted to use `static`, but that doesn't work yet it seems.